### DISCUSSION TITLE
extracting vertical level from NEMO output

### DISCUSSION THREAD

**Discussion by jupalm** [SciTools/iris#5149] [Q&A] [answered]
Hi everyone, 

I am using iris with NEMO outputs, and always get stuck when trying to extract a cube with iris tools.
i know i cannot do anything with latitude or longitude because of the non-regular NEMO grid, but i should be able to use iris tools to extract a vertical level, a specific depth.
i managed to get something working by reading the 3D field i want to plot, read a 2D field to use its 2D cube, and copy the vertical level i'm interested in into the 2D cube. It works, but it's far from being the most efficient way of doing.

my problem is the name of the Depth dimension , here is what a 3D cube from NEMO-MEDUSA looks like: 
```
dissolved inorganic carbon / (mmol-C/m3) (time: 1; Vertical T levels: 75; -- : 332; -- : 362)
    Dimension coordinates:
        time                                  x                     -        -         -
        Vertical T levels                     -                     x        -         -
    Auxiliary coordinates:
        time                                  x                     -        -         -
        latitude                              -                     -        x         x
        longitude                             -                     -        x         x
    Cell methods:
        mean                             time (2700 s)
        mean                             time
    Attributes:
        Conventions                      'CF-1.6'
        NCO                              'netCDF Operators version 4.7.6 (Homepage = http://nco.sf.net/, Code = h ...'
        description                      'Medusa sms variables'
        history                          'Tue Feb  1 09:32:15 2022: ncra eORCA1_MED_UKESM_1y_20000101_20011230_ptrc_T.nc ...'
        interval_operation               '2700 s'
        interval_write                   '1 yr'
        name                             'eORCA1_MED_UKESM_1y_20000101_20011230_ptrc_T'
        online_operation                 'average'
        timeStamp                        '2022-Jan-23 00:50:09 GMT'
        title                            'Medusa sms variables'
        uuid                             '4683442e-1f36-44c1-ad6f-1cb781007da2'
```

Do you have an idea how to extract the surface field for example, using iris.Extract and iris.Constraint ?

Huge thanks in advance for your help 

Cheers, 

Julien

**Comment by trexfeathers** [comment, marked as answer]
Hi @jupalm, thanks for getting in touch. We believe this is due to the spaces in `Vertical T levels`. Try using `coord_values`, as documented [here](https://scitools-iris.readthedocs.io/en/latest/generated/api/iris.html#iris.Constraint).

**Reply by jupalm** [reply]
Thanks! 
You were right,
i managed to extract the surface field like this : 
```
surf_slice = cube.extract(iris.Constraint(coord_values={'Vertical T levels':lambda cell: cell == 0}))
```
I it works fine.

Many thanks!

Julien.
